File: hw/maisincho/u3/HW20Compiste/Composite/src/main/java/ec/edu/espe/composite/model/Supervisor.java
package ec.edu.espe.composite.model;

import ec.edu.espe.composite.controller.Employee;
import java.util.Vector;

/**
 *
 * @author Isaac Maisincho Crafters_Market DCCO ESPE
 */
public abstract class Supervisor extends Employee {

    protected Vector directReports = new Vector();

    @Override
    public void stateName() {
        super.stateName();
        if (directReports.size() > 0);
        for (int i = 0; i < directReports.size(); ++i) {
            ((Employee) directReports.elementAt(i)).stateName();
        }
    }
    @Override
    public void add(Employee anEmployee){
        this.directReports.addElement(anEmployee);
    }
}

File: hw/bonilla/U3/HW19Composite/Composite.js/model/Supervisor.js
const Employee = require('./Employee');

class Supervisor extends Employee {
  constructor() {
    super();
    this.directReports = [];
  }

  stateName() {
    super.stateName();
    if (this.directReports.length > 0) {
      this.directReports.forEach(emp => emp.stateName());
    }
  }

  add(anEmployee) {
    this.directReports.push(anEmployee);
  }
}

module.exports = Supervisor;
